tree/master/cloud/src/solution/Microsoft.Legal.MatterCenter.Service/Microsoft.Legal.MatterCenter.Repository/Matter/MatterRepository.cs

```diff
@@ -1177,7 +1187,11 @@ public bool SetItemPermission(ClientContext clientContext, IList<IList<string>>
                         {
                             stampedUserList[itemsToRemoveStampedUserProp[i]] = string.Empty;
                             stampedUserEmailsList[itemsToRemoveStampedUserProp[i]] = string.Empty;
-                            stampedRolesList[itemsToRemoveStampedUserProp[i]] = string.Empty;
+
+                            if (stampedRolesList != null && stampedRolesList.Count > 0 && stampedRolesList.IndexOf(itemsToRemoveStampedUserProp[i].ToString()) > 0)
```

Comment:
To handle index out of range exception for matters where there are no roles present